### PR TITLE
Convert HttpHeaders to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public Factory(long nowMillis, Request request, Response cacheResponse) {
           } else if ("ETag".equalsIgnoreCase(fieldName)) {
             etag = value;
           } else if ("Age".equalsIgnoreCase(fieldName)) {
-            ageSeconds = HttpHeaders.parseSeconds(value, -1);
+            ageSeconds = UtilKtKt.toNonNegativeInt(value, -1);
```
Comment (oldergod) [inline okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.java]: `UtilKtKt` isn't a great name.

File: okhttp/src/main/java/okhttp3/Cache.kt [hunk 2]
```diff
@@ -736,5 +737,77 @@ class Cache internal constructor(
...
+    /**
+     * Returns true if none of the Vary headers have changed between [cachedRequest] and
+     * [newRequest].
+     */
+    fun varyMatches(
+      cachedResponse: Response,
+      cachedRequest: Headers,
+      newRequest: Request
+    ): Boolean {
+      for (field in cachedResponse.headers().varyFields()) {
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/Cache.kt]: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/none.html

File: okhttp/src/main/java/okhttp3/Cache.kt [hunk 3]
```diff
@@ -736,5 +737,77 @@ class Cache internal constructor(
...
+        if (!"Vary".equals(name(i), ignoreCase = true)) {
+          continue
+        }
+
+        val value = value(i)
+        if (result.isEmpty()) {
+          result = TreeSet(String.CASE_INSENSITIVE_ORDER)
+        }
+        for (varyField in value.split(',')) {
+          (result as MutableSet<String>).add(varyField.trim())
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Cache.kt]: Ooooh I like that.

[the remark this answers is not in the document]
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Cache.kt]: Fixed

File: okhttp/src/test/java/okhttp3/HeadersTest.java [hunk 4]
```diff
@@ -440,7 +440,7 @@
         .add("WWW-Authenticate", "Digest realm=\"myrealm\", nonce=\"fjalskdflwejrlaskdfjlaskdjflaks"
             + "jdflkasdf\", qop=\"auth\", stale=\"FALSE\"")
         .build();
-    List<Challenge> challenges = HttpHeaders.parseChallenges(headers, "WWW-Authenticate");
+    List<Challenge> challenges = HttpHeadersKt.parseChallenges(headers, "WWW-Authenticate");
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply okhttp/src/test/java/okhttp3/HeadersTest.java]: Good call. Done.

File: okhttp/src/main/java/okhttp3/Cache.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Cache.kt]: Screen Shot 2019-05-06 at 8 15 37 PM

[the remark this answers is not in the document]
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/Cache.kt]: IntelliJ is smart

File: okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.java (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.java]: Give that one a minute; it’s callers will soon all be Kotlin.

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.java]: Fixed to be only one Kt.

**Pull request merged**